pkg/network/proxy/proxy.go

```diff
@@ -229,83 +265,135 @@ func (proxy *OsdnProxy) updateEgressNetworkPolicy(policy networkv1.EgressNetwork
 
 	// Add/Update/Delete firewall rules for the namespace
 	if len(firewall) > 0 {
-		if _, ok := proxy.firewall[ns]; !ok {
-			item := &proxyFirewallItem{}
-			item.namespaceFirewalls = make(map[ktypes.UID][]firewallItem)
-			item.activePolicy = nil
-			proxy.firewall[ns] = item
-		}
-		proxy.firewall[ns].namespaceFirewalls[policy.UID] = firewall
-	} else if _, ok := proxy.firewall[ns]; ok {
-		delete(proxy.firewall[ns].namespaceFirewalls, policy.UID)
-		if len(proxy.firewall[ns].namespaceFirewalls) == 0 {
-			delete(proxy.firewall, ns)
-		}
+		ns.firewalls[policy.UID] = firewall
+	} else {
+		delete(ns.firewalls, policy.UID)
```

Comment:
While you're at it, do you want to move the key from UID to NamespacedName? If an object is deleted-and-recreated with the same name, we could leak objects.

Reply:
I agree this code is weird - if you don't want to touch it, that's OK.